src/main/java/com/ironcorelabs/tenantsecurity/kms/v1/TenantSecurityClient.java

```diff
@@ -403,6 +293,48 @@ public static boolean isCiphertext(byte[] bytes) {
                 }));
     }
 
+    /**
+     * Encrypt the bytes in input and write it to output. A new key will be wrapped
+     * using the TSP. The returned value contains the edek needed to decrypt the
+     * resulting stream.
+     * 
+     * @param input    The input stream of bytes to encrypt.
+     * @param output   The output stream to write encrypted bytes to.
+     * @param metadata Metadata about the document being encrypted.
+     * @return The edek which can be used to decrypt the resulting stream
+     */
+    public CompletableFuture<StreamingResponse> encryptStream(InputStream input, OutputStream output,
+                    DocumentMetadata metadata) {
+            return this.encryptionService.wrapKey(metadata).thenApplyAsync(
+                            wrapResponse -> CryptoUtils
+                                            .encryptStreamInternal(wrapResponse.getDekBytes(), metadata, input, output,
+                                                            this.secureRandom)
+                                            .thenApply(unused -> new StreamingResponse(wrapResponse.getEdek())).join(),
+                            encryptionExecutor);
+    }
+
+    /**
+     * Decrypt the bytes that are represented by input using the key contained
+     * inside the edek. Unverified bytes will be written to the output stream. Once
+     * the GCM tag has been reached and verified, this function will return. If
+     * there is a problem with the document represented by input or a problem
+     * unwrapping the edek the returned CompletableFuture will return an exception
```

Comment:
Should we communicate more about the types of exceptions that can happen here?

Reply:
I didn't since we don't on other methods. What all would you like to see communicated?

Reply:
I was thinking whatever would be in the `@throws` for a normal method so the caller can know what to expect. If this is how the public API is in other places, we should probably remain consistent and change them all at once if we want to do that.